pkg/lib/accountmanagement/token.go

```diff
@@ -9,22 +9,33 @@ import (
 )
 
 type Token struct {
-	AppID       string     `json:"app_id,omitempty"`
-	UserID      string     `json:"user_id,omitempty"`
-	Alias       string     `json:"alias,omitempty"`
-	RedirectURI string     `json:"redirect_uri,omitempty"`
-	State       string     `json:"state,omitempty"`
-	TokenHash   string     `json:"token_hash,omitempty"`
-	CreatedAt   *time.Time `json:"created_at,omitempty"`
-	ExpireAt    *time.Time `json:"expire_at,omitempty"`
+	AppID     string     `json:"app_id,omitempty"`
+	UserID    string     `json:"user_id,omitempty"`
+	TokenHash string     `json:"token_hash,omitempty"`
+	CreatedAt *time.Time `json:"created_at,omitempty"`
+	ExpireAt  *time.Time `json:"expire_at,omitempty"`
+
+	// Adding OAuth
+	Alias       string `json:"alias,omitempty"`
+	RedirectURI string `json:"redirect_uri,omitempty"`
+	State       string `json:"state,omitempty"`
+
+	// Adding Identity
+	Identity *IdentityToken `json:"identity_token,omitempty"`
+}
+
+type IdentityToken struct {
+	IdentityID  string `json:"identity_id,omitempty"`
+	PhoneNumber string `json:"phone_number,omitempty"`
+	Email       string `json:"email,omitempty"`
 }
 
 func (t *Token) CheckUser(userID string) error {
 	if subtle.ConstantTimeCompare([]byte(t.UserID), []byte(userID)) == 1 {
 		return nil
 	}
 
-	return ErrOAuthTokenNotBoundToUser
+	return ErrAccountManagementTokenNotBoundToUser
```

Comment:
This line changes the API accidentally. We should keep the existing OAuth API to return the same error.

We can keep the compatibility by adding CheckUser_OAuth and CheckUser. Change the OAuth code to use CheckUser_OAuth.

Reply:
We've kept the api by checking err if `ErrAccountManagementTokenNotBoundToUser` and then rethrow `ErrOAuthTokenNotBoundToUser`.

Anyway, let me change to implement `CheckUser_OAuth` instead